### PR TITLE
Recommendation1

### PR DESCRIPTION
**Pull request by Oluwatosin12-bit**
This PR highlights the following:
- users being able to delete posts that they created
- users have their likes stored for references and the data of all liked posts is displayed on another page 
- recommendation system for each user is more robust. It includes categories (as well as related categories) from posts they like or book an appointment for.

https://www.loom.com/share/6351369d1bec41eab663496736af67c7?sid=b742a6b2-ba9f-4000-af24-a3b097febd8b

### CODE REVIEW

File: FrontEnd/src/ProfilePage/UserProfile.jsx [hunk 1]
```diff
@@ -39,15 +39,15 @@ function UserProfilePage({ userUID, userData }) {
...
+      signOutDropdownRef.current !==null &&
       !signOutDropdownRef.current.contains(event.target)
     ) {
       setisSignOutDropdownVisible(false);
     }
 
     for (let postId in deletePostDropdownRef.current) {
       if (
-        deletePostDropdownRef.current[postId] &&
+        deletePostDropdownRef.current[postId] !==null &&
```
Comment (pearlor) [inline FrontEnd/src/ProfilePage/UserProfile.jsx:50]: Spacing `!== null`

File: FrontEnd/src/ProfilePage/UserProfile.jsx [hunk 2]
```diff
@@ -39,15 +39,15 @@ function UserProfilePage({ userUID, userData }) {
 
   const handleClickOutside = (event) => {
     if (
-      signOutDropdownRef.current &&
+      signOutDropdownRef.current !==null &&
```
Comment (pearlor) [inline FrontEnd/src/ProfilePage/UserProfile.jsx:42]: Spacing `!== null`

File: FrontEnd/src/Header.jsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import "./Header.css";
 import NotificationsPage from "./Notifications/NotificationsPage";
 function Header({ userData, socket }) {
   const [notifications, setNotifications] = useState([]);
-  const [openNotifications, setOpenNotifications] = useState(true);
+  const [isNotificationsOpen, setisNotificationsOpen] = useState(true);
```
Comment (pearlor) [inline FrontEnd/src/Header.jsx:8]:
```suggestion
  const [isNotificationsOpen, setIsNotificationsOpen] = useState(true);
```

File: FrontEnd/src/UserAuthentication/LandingPage.jsx [hunk 4]
```diff
@@ -4,24 +4,23 @@ import "./LandingPage.css";
 
 function LandingPage() {
   const { theme, setTheme } = useTheme();
-  const [menuOpen, setMenuOpen] = useState(false);
-
+  const [ismenuOpen, setIsMenuOpen] = useState(false);
```
Comment (pearlor) [inline FrontEnd/src/UserAuthentication/LandingPage.jsx:7]:
```suggestion
  const [isMenuOpen, setIsMenuOpen] = useState(false);
```

File: FrontEnd/src/ProfilePage/UserProfile.jsx [hunk 5]
```diff
@@ -17,15 +21,37 @@ function UserProfilePage({ userUID, userData }) {
   const [isFormValid, setIsFormValid] = useState(false);
   const [serviceCategories, setSelectedCategories] = useState([]);
   const [availableCategories] = useState(CATEGORIES);
-  const [isDropdownVisible, setDropdownVisible] = useState(false);
-  const dropdownRef = useRef(null);
+  const [isSignOutDropdownVisible, setisSignOutDropdownVisible] =
+    useState(false);
+  const [isDeletePostDropdownVisible, setisDeletePostDropdownVisible] =
```
Comment (pearlor) [inline FrontEnd/src/ProfilePage/UserProfile.jsx:26]:
```suggestion
  const [isDeletePostDropdownVisible, setIsDeletePostDropdownVisible] =
```

File: FrontEnd/src/ProfilePage/UserProfile.jsx [hunk 6]
```diff
@@ -17,15 +21,37 @@ function UserProfilePage({ userUID, userData }) {
   const [isFormValid, setIsFormValid] = useState(false);
   const [serviceCategories, setSelectedCategories] = useState([]);
   const [availableCategories] = useState(CATEGORIES);
-  const [isDropdownVisible, setDropdownVisible] = useState(false);
-  const dropdownRef = useRef(null);
+  const [isSignOutDropdownVisible, setisSignOutDropdownVisible] =
```
Comment (pearlor) [inline FrontEnd/src/ProfilePage/UserProfile.jsx:24]:
```suggestion
  const [isSignOutDropdownVisible, setIsSignOutDropdownVisible] =
```

File: FrontEnd/src/ProfilePage/UserProfile.jsx [hunk 7]
```diff
@@ -17,15 +21,37 @@ function UserProfilePage({ userUID, userData }) {
...
+  const deletePostDropdownRef = useRef({});
 
-  const toggleDropdown = () => {
-    setDropdownVisible(!isDropdownVisible);
+  const toggleSignOutDropdown = () => {
+    setisSignOutDropdownVisible(!isSignOutDropdownVisible);
   };
+  const toggleDeletePostDropdown = (index) => {
+    setisDeletePostDropdownVisible(
+      isDeletePostDropdownVisible === index ? null : index
```
Comment (pearlor) [inline FrontEnd/src/ProfilePage/UserProfile.jsx:36]: Why is `setisDeletePostDropdownVisible ` being set with null or index? Shouldn't it be false or true based on the name of the variable and setter?

Comment (Oluwatosin12-bit) [reply FrontEnd/src/ProfilePage/UserProfile.jsx:36]: There are multiple posts that are mapped, that's how I am able to differentiate which one of them to delete

**Pull request merged**